go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -188,16 +191,14 @@ def in_temp_dir(stub_paths, &block)
         # the layout of the filesystem with a structure we can reproduce (i.e.
         # no paths such as ../../../foo), run the Go tooling, then reverse the
         # process afterwards.
-        def replace_directive_substitutions(go_mod_content)
+        def replace_directive_substitutions
           @replace_directive_substitutions ||=
-            SharedHelpers.in_a_temporary_directory do |path|
-              File.write("go.mod", go_mod_content)
-
+            begin
               # Parse the go.mod to get a JSON representation of the replace
               # directives
               command = "go mod edit -json"
```

Comment:
Nit: iff the ` parse_manifest_requirements` comment above, it could be reused here too.

Reply:
Nit: it's weird we're forking `go mod edit -json` at least 3 times during the update process...

AFAICT this is using the same output as we use for `original_paths` calculation and could be shared?